lib/healthcheck/report.rb

```diff
@@ -27,14 +27,15 @@ def to_json(_ = nil)
 
     private
 
-    def initialize_check(check)
+    def initialize_check(check, query)
       case check
       when Healthcheck::Checks::AbstractCheck then check
       when Class then check.new
       else check.constantize.new
       end.tap do |c|
         c.reset
         c.paused = paused_data[c.class.slug]
+        c.set_filter_by_query(query) unless query.nil?
```

Comment:
I think this could be `c.set_filter_by_query(query) if query.present?` so we don't apply the filter for null or empty hashes.

Reply:
👍 sounds better. Thanks